### PR TITLE
PPU: improve accuracy for instructions FRSQRTE, FRES

### PR DESCRIPTION
**Pull request by Nekotekina**
FRSQRTE implementation is based on #7135 PR.
Don't forget to clear LLVM Cache to take effect.
FRES should be accurate, but for FRSQRTE some special values, especially denormals, don't produce accurate results due to high complexity of handling denormals.

### CODE REVIEW

**Comment by Nekotekina** [comment]
Made some hw testing. Denormals are cursed, probably quasi-accurate implementation should suffice.

**Comment by Nekotekina** [comment]
Implemented FRES instruction with a lookup table.

File: rpcs3/Emu/Cell/PPUInterpreter.cpp [hunk 1]
```diff
@@ -6176,9 +6176,16 @@ auto FRES()
...
-	ppu_set_fpcc<Flags...>(ppu, ppu.fpr[op.frd], 0.);
+	static const auto exec = [](ppu_thread& ppu, ppu_opcode_t op)
+	{
+		const f64 a = ppu.fpr[op.frb];
+		const u64 b = std::bit_cast<u64>(a);
+		const u64 e = (b >> 52) & 0x7ff; // double exp
+		const u64 i = (b >> 45) & 0x7f; // mantissa LUT index
+		const u64 r = e >= (0x3ff + 0x80) ? 0 : (0x7ff - 2 - e) << 52 | u64{ppu_fres_mantissas[i]} << (32 - 3);
+
+		ppu.fpr[op.frd] = f32(std::bit_cast<f64>(a == a ? (b & 0x8000'0000'0000'0000) | r : (0x8'0000'0000'0000 | b)));
```
Comment (Firnis) [inline rpcs3/Emu/Cell/PPUInterpreter.cpp:6187]: `a == a` seems like a typo

Comment (Nekotekina) [reply rpcs3/Emu/Cell/PPUInterpreter.cpp:6187]: No, it's testing for NaN.

**Pull request merged**